extensions/default/src/commandsModule.ts

```diff
@@ -279,38 +280,50 @@ const commandsModule = ({
     toggleHangingProtocol: ({
       protocolId,
       stageIndex,
+      displaySetSelectorMap,
+      toggledState,
     }: HangingProtocolParams): boolean => {
       const {
         protocol,
         stageIndex: desiredStageIndex,
         activeStudy,
       } = hangingProtocolService.getActiveProtocol();
       const { toggleHangingProtocol } = stateSyncService.getState();
-      const storedHanging = `${
-        activeStudy.StudyInstanceUID
-      }:${protocolId}:${stageIndex | 0}`;
+      // Just use the protocol id as the toggle id for toggling
+      // as the stage shouldn't matter, and one wants to restore even across
+      // active studies.
+      const toggleID = protocolId;
```

Comment:
It is now possible to toggle on a hanging protcool with multiple stages, and then it won't toggle off until a different hanging protocol is applied or the toggle is directly called.  That allows for toggling on a mode such as MPR, and having commands which either apply a given stage or toggle the stage on if it is off, and those same commands will toggle the HP off.